### ISSUE TITLE
Better support for downloading binary content

### ISSUE THREAD

**Issue by cjstehno** [cjstehno/ersatz#91]
Test scenario: 

Download a zip file from server and open with ZipFile and read contents.
The file should be downloaded with a GET and should have

```
content(data.bytes, 'application/zip')
header('Content-Disposition', 'attachment; filename="data.zip"')
```

This seems overly difficult to implement with existing code.

**Comment by cjstehno** [comment]
Supporting this will entail some deep changes in the way I handle the response content - it is for the better, but it will probably modify API interfaces unless I can come up with some means of supporting a graceful transition. I am shelving this until 1.8 or 2.0 release.

Some notes are in the `response-bytes` branch.

**Comment by cjstehno** [comment]
This will need to be part of 2.0 as originally specified since I will need to change the internal response storage to a byte array.

**Comment by cjstehno** [comment]
This has been implemented in v2.0, which is the next release.
